daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLAppender.scala

```diff
@@ -38,25 +38,32 @@ class TDMLAppender
     import TDMLAppender._
 
     override def append(event: LogEvent): Unit = {
-        diagnostics = new LogDiagnostic(event.getMessage.getFormattedMessage) +: diagnostics
+        checkForKey
+        diagnostics = diagnostics.updated(Thread.currentThread.getId, diagnostics(Thread.currentThread.getId) :+ new LogDiagnostic(event.getMessage.getFormattedMessage))
     }
 }
 
 object TDMLAppender{
-    var diagnostics: Seq[Diagnostic] = Nil
-    var diagnostics2: Seq[String] = Nil
+    var diagnostics: Map[Long, Seq[Diagnostic]] = Map()

```

Comment:
Does it make sense to use a mutable map and make this a val?

Reply:
Sure, but that doesn't change the fact that a singleton object has mutable state in it. I think this appender needs to be an allocated object, not a singleton.